options/options.go

```diff
@@ -452,6 +454,61 @@ func (opts *ToolOptions) ParseArgs(args []string) ([]string, error) {
 	return args, err
 }
 
+// LogSensitiveOptionWarnings logs a warning for any sensitive information (i.e. passwords)
+// that appear on the command line for the --password, --uri and --sslPEMKeyPassword options.
+// This also applies to a connection string that appears as a positional argument.
+func LogSensitiveOptionWarnings(args []string) {
+	passwordMsg := "WARNING: On some systems, a password provided directly using " +
+		"--password may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	uriMsg := "WARNING: On some systems, a password provided directly using " +
+		"--uri may be visible to system status programs such as `ps` that may be " +
+		"invoked by other users. Consider omitting the password to provide it via stdin, " +
+		"or using the --config option to specify a configuration file with the password."
+
+	sslMsg := "WARNING: On some systems, a password provided directly using --sslPEMKeyPassword " +
+		"may be visible to system status programs such as `ps` that may be invoked by other users. " +
+		"Consider using the --config option to specify a configuration file with the password."
+
+	for i, arg := range args {
+		// Option has the form --password= but is not followed by the empty string.
+		if strings.HasPrefix(arg, "--password=") && len(arg) > 11 && arg[11:] != "" {
+			log.Logvf(log.Always, passwordMsg)
+			continue
+		}
+
+		// Option has the form --password and the next argument is not the empty string.
+		if strings.HasPrefix(arg, "--password") && i+1 < len(args) && args[i+1] != "" {
+			log.Logvf(log.Always, passwordMsg)
+			continue
+		}
+		// Option has the form --uri= with at least one character after it.
+		if strings.HasPrefix(arg, "--uri=") && len(arg) > 6 {
+			if cs, err := connstring.Parse(arg[6:]); err == nil && cs.Password != "" {
```

Comment:
I originally thought we could check `cs.PasswordSet == true`, but that gets set to true as long as there's a colon after the username in the connection string [according to the driver code](https://github.com/mongodb/mongo-tools/blob/22bb313a5a5c4e1572b3082b8afb55efd00f3919/vendor/go.mongodb.org/mongo-driver/x/mongo/driver/connstring/connstring.go#L216) (even if the password is blank).

So if you have `mongodb://abc:@localhost:33333`, the driver would think the password is set. Not sure why that's the case since passwords can't be blank afaik, but checking `cs.Password != ""` is basically the same thing.